runtime/src/qdrant.rs

```diff
@@ -0,0 +1,122 @@
+﻿use std::collections::HashMap;
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use log::{debug, error, info, warn};
+use once_cell::sync::Lazy;
+use rocket::get;
+use tauri::api::process::{Command, CommandChild, CommandEvent};
+use tauri::Url;
+use crate::api_token::{APIToken};
+use crate::environment::DATA_DIRECTORY;
+
+// Qdrant server process started in a separate process and can communicate
+// via HTTP or gRPC with the .NET server and the runtime process
+static QDRANT_SERVER: Lazy<Arc<Mutex<Option<CommandChild>>>> = Lazy::new(|| Arc::new(Mutex::new(None)));
+
+// Qdrant server port (default is 6333 for HTTP and 6334 for gRPC)
+static QDRANT_SERVER_PORT: Lazy<u16> = Lazy::new(|| {
+    crate::network::get_available_port().unwrap_or(6333)
+}); 
+
+static QDRANT_INITIALIZED: Lazy<Mutex<bool>> = Lazy::new(|| Mutex::new(false));
+
+#[get("/system/qdrant/port")]
+pub fn qdrant_port(_token: APIToken) -> String {
+    let qdrant_port = *QDRANT_SERVER_PORT;
+    format!("{qdrant_port}")
+}
+
+/// Starts the Qdrant server in a separate process.
+pub fn start_qdrant_server() {
+    
+    let base_path = DATA_DIRECTORY.get().unwrap(); 
+    
+    let storage_path = Path::new(base_path).join("databases").join("qdrant").join("storage").to_str().unwrap().to_string();
+    let snapshot_path = Path::new(base_path).join("databases").join("qdrant").join("snapshots").to_str().unwrap().to_string();
+    let init_path = Path::new(base_path).join("databases").join("qdrant").join(".qdrant-initalized").to_str().unwrap().to_string();
+    
+    println!("{}", storage_path);
+    println!("{}", snapshot_path);
+    println!("{}", init_path);
```

Comment:
Printing makes little sense in release. Should we log these values as info instead?